sdks/python/apache_beam/runners/direct/executor.py

```diff
@@ -291,44 +294,62 @@ def call(self):
 
     side_input_values = [self._side_input_values[side_input]
                          for side_input in self._applied_ptransform.side_inputs]
-
-    try:
-      evaluator = self._transform_evaluator_registry.get_evaluator(
-          self._applied_ptransform, self._input_bundle,
-          side_input_values, scoped_metrics_container)
-
-      if self._fired_timers:
-        for timer_firing in self._fired_timers:
-          evaluator.process_timer_wrapper(timer_firing)
-
-      if self._input_bundle:
-        for value in self._input_bundle.get_elements_iterable():
-          evaluator.process_element(value)
-
-      with scoped_metrics_container:
-        result = evaluator.finish_bundle()
-        result.logical_metric_updates = metrics_container.get_cumulative()
-
-      if self._evaluation_context.has_cache:
-        for uncommitted_bundle in result.uncommitted_output_bundles:
+    _unprocessed = True
+    while (_unprocessed and
+           self._retry_count < TransformExecutor._MAX_RETRY_PER_BUNDLE):
+      try:
+        self.attempt_call(metrics_container,
+                          scoped_metrics_container,
+                          side_input_values)
+        _unprocessed = False
```

Comment:
<!--new_thread; commit:d61a79d86a74422660f4959a8abb348f9f1d5129; resolved:0-->
You can get rid of the `_unprocessed` variable and just use a `break` here.